### ISSUE TITLE
Example project structure.

### ISSUE THREAD

**Issue by seanparsons** [sseefried/docker-build-ghc-android#5]
While this docker image gets people a few steps forward, it would be useful to provide a sample project that people can use to springboard off into their own work.

**Comment by sseefried** [comment]
Hi Sean,

That's a great idea. I'll try to add soon.

Sean

**Comment by seanparsons** [comment]
Any thoughts on this? If you've got any starting points I'll be happy to give them a bash.

**Comment by sseefried** [comment]
Perhaps just a very simple Hello World app would be good first. They can "adb shell" in and see that it is working.

But in the long run I suppose it would be nice to have some kind of app. The question then becomes, is this going to be a Java app? To tell the truth the only app I have built for so far is a native SDL app (with a small Java shim) but I think that's too complicated for a newbie.

Perhaps we could try using this? https://github.com/neurocyte/android-haskell-activity

What are your thoughts?

**Comment by seanparsons** [comment]
I wasn't sure about that because of the need to install the foreign-jni package and the like which seems like an unnecessary faff.

**Comment by sseefried** [comment]
Yes, I was a bit worried about that too. I expect, however, that some people would really like to see a proper application. What about a simple OpenGL graphic effect or the like? Could probably do that by calling Haskell from a small Java program, but write the OpenGL effect in Haskell.

**Comment by seanparsons** [comment]
That would be perfect.

**Comment by seanparsons** [comment]
How about this as a starting point for the code?
http://dpwright.com/posts/2015/03/25/the-haskell-gl-package/

**Comment by johnjelinek** [comment]
@sseefried: I built your docker image last night and this morning I wanted to give it a try. From initial research, I found the same https://github.com/neurocyte/android-haskell-activity app you mentioned above. I'd really like to start with this app, but ass @seanparsons mentioned, it requires foreign-jni, which I've had some trouble building:

```
androidbuilder@012b479ea572:~/foreign-jni$ ~/.ghc/android-14/arm-linux-androideabi-4.8/bin/arm-linux-androideabi-cabal update
Downloading the latest package list from hackage.haskell.org
androidbuilder@012b479ea572:~/foreign-jni$ ~/.ghc/android-14/arm-linux-androideabi-4.8/bin/arm-linux-androideabi-cabal install
Resolving dependencies...
Downloading text-1.2.0.4...
Downloading transformers-0.4.3.0...
Configuring transformers-0.4.3.0...
Configuring text-1.2.0.4...
Failed to install transformers-0.4.3.0
Build log ( /home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/.cabal/logs/transformers-0.4.3.0.log ):
/home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/cabal/setup-exe-cache/setup-Simple-Cabal-1.18.1.3-arm-linux-android-ghc-7.8.3: 1: /home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/cabal/setup-exe-cache/setup-Simple-Cabal-1.18.1.3-arm-linux-android-ghc-7.8.3: Syntax error: word unexpected (expecting ")")
[1 of 1] Compiling Main             ( /tmp/transformers-0.4.3.0-343/transformers-0.4.3.0/dist/setup/setup.hs, /tmp/transformers-0.4.3.0-343/transformers-0.4.3.0/dist/setup/Main.o )
Linking /tmp/transformers-0.4.3.0-343/transformers-0.4.3.0/dist/setup/setup ...
Failed to install text-1.2.0.4
Build log ( /home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/.cabal/logs/text-1.2.0.4.log ):
/home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/cabal/setup-exe-cache/setup-Simple-Cabal-1.18.1.3-arm-linux-android-ghc-7.8.3: 1: /home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/cabal/setup-exe-cache/setup-Simple-Cabal-1.18.1.3-arm-linux-android-ghc-7.8.3: Syntax error: word unexpected (expecting ")")
cabal: Error: some packages failed to install:
foreign-jni-0.1.0.0 depends on text-1.2.0.4 which failed to install.
mtl-2.2.1 depends on transformers-0.4.3.0 which failed to install.
text-1.2.0.4 failed during the configure step. The exception was:
ExitFailure 2
transformers-0.4.3.0 failed during the configure step. The exception was:
ExitFailure 2
```

I think it'd be valuable to make a new tag/branch of the docker container with whatever java requirements are necessary to get that going. If you'd like, I can try to extend it with a JDK and adding some env variables like `JAVA_HOME` if you think it'll be helpful.

**Comment by sseefried** [comment]
@johnjelinek, sorry for being so tardy on this issue.

The problem is the line 

jobs: $ncpus

in /home/androidbuilder/.ghc/android-14/arm-linux-androideabi-4.8/.cabal/config

**Comment by sseefried** [comment]
Fixed in https://github.com/sseefried/docker-build-ghc-android/commit/e136d7f9063f65ef74145abce000df28addca785

**Comment by sseefried** [comment]
There is now a project built on top of this that builds a simple Android game.

https://github.com/sseefried/docker-epidemic-build-env

**Comment by seanparsons** [comment]
@sseefried You sir are a champion!

**Comment by sseefried** [comment]
Thanks :-) It's take a while to get all this working, but it was worth it.
